### PR TITLE
complete forget password flow

### PR DESCRIPTION
**Pull request by Ishaan-11**

### CODE REVIEW

File: api/routes/forgot-password/controllers.js [hunk 1]
```diff
@@ -0,0 +1,74 @@
+const BaseController = require('base/controllers/baseController');
+const DB = require('models');
+const config = require('config');
+const ApiError = require('base/error');
+
+class ForgetPasswordController extends BaseController {
+  async post() {
+    const { email } = this.request.body;
```
Comment (jatinkatyal13) [inline api/routes/forgot-password/controllers.js:8]: we can get the user from `this.request.user`

File: api/routes/forgot-password/controllers.js [hunk 2]
```diff
@@ -0,0 +1,74 @@
...
+    await _resetPasswordService.verifyResetToken(token);
+
+    this.response.status(204).send({});
+  }
+
+  async postResetPassword() {
+    const { token } = this.request.params;
+    const { password, passwordRepeat } = this.request.body;
+
+    if (password !== passwordRepeat) {
```
Comment (jatinkatyal13) [inline api/routes/forgot-password/controllers.js]: this should ideally be in schema or validation preHandler

File: api/routes/forgot-password/controllers.js (hunk 1 above)
Comment (Ishaan-11) [reply api/routes/forgot-password/controllers.js:8]: i think user won't be logged in, in this flow

Comment (jatinkatyal13) [reply api/routes/forgot-password/controllers.js:8]: sorry yes !

**Review by jatinkatyal13 — APPROVED** [review]

**Pull request merged**